### PR TITLE
Add option to provide the RPC URL for a localhost network deployment

### PR DESCRIPTION
**Pull request by stubbsta**
This PR adds a feature required by the waku-simulator to enable deployment to a network other than sepolia or localhost.

1. RPC_PROVIDER environment variable added
2. If the RPC_PROVIDER variable is provided then it is assigned as the URL for the localhost_integration network in hardhat.config.ts

### CODE REVIEW

**Review by alrevuelta — APPROVED** [review]
lgtm. Lets wait for @rymnc review.

File: hardhat.config.ts
Comment (stubbsta) [reply hardhat.config.ts]: something like this [2354961](https://github.com/waku-org/waku-rln-contract/pull/13/commits/23549614cafee588c127f15f3ea09adf0e0f0187)?

**Review by rymnc — APPROVED** [review]
LGTM, can be merged after @stubbsta confirms it works with waku-simulator

**Comment by stubbsta** [comment]
This change isn't working when testing in waku-simulator.
I looks like the config from hardhat.config.ts is ignored, because the error shows the localhost IP of 127.0.0.1:
```
HardhatError: HH108: Cannot connect to the network localhost.
Please make sure your node is running, and check your internet connection and networks config
    at HttpProvider._fetchJsonRpcResponse (/waku-rln-contract/node_modules/hardhat/src/internal/core/providers/http.ts:221:15)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async HttpProvider.request (/waku-rln-contract/node_modules/hardhat/src/internal/core/providers/http.ts:85:29)
    at async DeploymentsManager.getChainId (/waku-rln-contract/node_modules/hardhat-deploy/src/DeploymentsManager.ts:443:23)
    at async DeploymentsManager.loadDeployments (/waku-rln-contract/node_modules/hardhat-deploy/src/DeploymentsManager.ts:619:17)
    at async DeploymentsManager.runDeploy (/waku-rln-contract/node_modules/hardhat-deploy/src/DeploymentsManager.ts:998:5)
    at async SimpleTaskDefinition.action (/waku-rln-contract/node_modules/hardhat-deploy/src/index.ts:438:5)
    at async Environment._runTaskDefinition (/waku-rln-contract/node_modules/hardhat/src/internal/core/runtime-environment.ts:330:14)
    at async Environment.run (/waku-rln-contract/node_modules/hardhat/src/internal/core/runtime-environment.ts:163:14)
    at async SimpleTaskDefinition.action (/waku-rln-contract/node_modules/hardhat-deploy/src/index.ts:584:32)

    Caused by: Error: connect ECONNREFUSED 127.0.0.1:8545
        at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1555:16)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Is there some hardhat setting required to ensure it uses the provided config?

**Comment by stubbsta** [comment]
Running 
```
yarn deploy localhost_integration
```
works and uses the provided URL.

Using the network name 'localhost' seems to bypass using the config from hardhat.config.ts.
With
```
yarn deploy localhost
```
I am able to successfully deploy the contracts from my local machine (not inside docker) with no .env file present or even with 
```
networks: {},
```
instead of 
```
networks: getNetworkConfig,
```

Is it acceptable to use "localhost_integration" as the network or do we want to find a way to use "localhost" with a given RPC URL?

**Comment by stubbsta** [comment]
We could also change the network name in hardhat.config.ts from 'localhost_integration' to 'localhost', then using
'''yarn deploy localhost''' works as expected

**Review by rymnc — APPROVED** [review]
LGTM

**Pull request merged**
